core/coreobjects/include/coreobjects/property_impl.h

```diff
@@ -114,6 +115,33 @@ class PropertyImpl : public ImplementationOf<IProperty, ISerializable, IProperty
         checkErrorInfo(validateDuringConstruction());
     }
 
+    PropertyImpl(IPropertyBuilder* propertyBuilder)
+    {
+        const auto propertyBuilderPtr = PropertyBuilderPtr(propertyBuilder);
+        this->valueType = propertyBuilderPtr.getValueType();
+        this->name = propertyBuilderPtr.getName();
+        this->description = propertyBuilderPtr.getDescription();
+        this->unit = propertyBuilderPtr.getUnit();
+        this->minValue = propertyBuilderPtr.getMinValue();
+        this->maxValue = propertyBuilderPtr.getMaxValue();
+        this->defaultValue = propertyBuilderPtr.getDefaultValue();
+        this->visible = propertyBuilderPtr.getVisible();
+        this->readOnly = propertyBuilderPtr.getReadOnly();
+        this->selectionValues = propertyBuilderPtr.getSelectionValues();
+        this->suggestedValues = propertyBuilderPtr.getSuggestedValues();
+        this->refProp = propertyBuilderPtr.getReferencedProperty();
+        this->coercer = propertyBuilderPtr.getCoercer();
+        this->validator = propertyBuilderPtr.getValidator();
+        this->callableInfo = propertyBuilderPtr.getCallableInfo();
+        this->onValueWrite = EventEmitter(propertyBuilderPtr.getOnPropertyValueWrite());
+        this->onValueRead = EventEmitter(propertyBuilderPtr.getOnPropertyValueRead());
```

Comment:
Is the EventEmitter wrapper required here?

Reply:
It shouldn't be necessary, otherwise the `operator=` needs to be fixed.